src/main/java/io/cdap/delta/bigquery/BigQueryAssessor.java

```diff
@@ -61,6 +65,20 @@ public TableAssessment assess(StandardizedTableDetail tableDetail) {
     List<Problem> problems = new ArrayList<>();
     String dbName = tableDetail.getDatabase();
     String tableName = tableDetail.getTable();
+
+
+    if (tables.containsKey(tableName)) {
+      // TODO support same table name in different schema
+      if (!Objects.equals(tables.get(tableName), tableDetail.getSchemaName())) {
+        problems.add(new Problem("Duplicate Table Name", String.format("Duplicated table name %s detected. BigQuery " +
+          "doesn't support duplicate table name even though they are in different schema", tableName),
+          "Please only select one of the tables with same table name to replicate",
+          "Not be able to replicate multiple tables with same name " + "to BigQuery"));
+      }
+      // else if schema name is same, that means it's assessing the same table.
+    } else {
+      tables.put(tableName, tableDetail.getSchemaName());
```

Comment:
This `else` block can be removed if we return from `if` on problem.

Reply:
this else corresponds to `if (tables.containsKey(tableName))`